### PR TITLE
Script to count tree sizes

### PR DESCRIPTION
**Pull request by nritsche**

### CODE REVIEW

File: scripts/dataset_utils [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.command()
+@click.argument("dataset_id")
+def treesize(dataset_id):
```
Comment (jrs65) [inline scripts/dataset_utils]: This should have a docstring such that it gives a help message saying what it does.

File: setup.py [hunk 2]
```diff
@@ -23,4 +24,5 @@
     description="CHIME dataset (comet) ORM",
     license="GPL v3.0",
     url="https://github.com/chime-experiment/chimedb_dataset",
+    scripts=["scripts/dataset_utils"],
```
Comment (jrs65) [inline setup.py]: This should use the click entrypoints method of creating the script.

**Review by jrs65 — CHANGES_REQUESTED** [review]

**Comment by nritsche** [comment]
Thanks for reviewing, @jrs65 . I think I made all the additions you requested.

File: chimedb/dataset/utils.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+@click.group()
+def cli():
+    """Click entry point."""
+    pass
+
+
+@cli.command()
+@click.argument("dataset_id")
+def treesize(dataset_id):
+    """Print number of nodes in a dataset tree."""
```
Comment (jrs65) [inline chimedb/dataset/utils.py]:
```suggestion
    """Print number of nodes in the tree containing DATASET_ID"""
```

Comment (jrs65) [reply chimedb/dataset/utils.py]: I think you're typically supposed to mention the arguments when writing docstrings that end up as click help messages.

File: chimedb/dataset/utils.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    while not node.root:
+        node = node.base_dataset
+        if node in tree:
+            return True
+
+    return False
+
+
+if __name__ == "__main__":
+    cli()
```
Comment (jrs65) [inline chimedb/dataset/utils.py]:
```suggestion
```
You could remove this if you don't intend for the script to be called directly.

**Review by jrs65 — APPROVED** [review]
A few small comments to think about, but do what you want with them.

**Pull request merged**